### ISSUE TITLE
using a filter to not show all images using isotope on page load

### ISSUE THREAD

**Issue by leelnj** [metafizzy/isotope#450]
I have the following html (put html code which is relevant) and Jquery code (jquery code to long to input) which creates an isotype filter within my website on gallery images. Its working fine but when the page loads it shows all the images and i only want it to show the design images. I thought i would have to ammend the filter in the jquery code from '*' to '.design' but its not working. Any ideas thanks in advance

HTML

```
<section id="options" class="clearfix">
                <ul id="filters" class="option-set clearfix" data-option-key="filter">
                    <li><a href="#filter" data-option-value="*" class="selected">ALL</a></li>
                    <li><a href="#filter" data-option-value=".design" class="selected">MODELLING</a></li>
                    <li><a href="#filter" data-option-value=".website">COMMERCIAL</a></li>
                    <li><a href="#filter" data-option-value=".photoshop">DANCE</a></li>
                    <li><a href="#filter" data-option-value=".tvfilm">TV &amp; FILM</a></li>
                </ul>
            </section>
        </div>
    </div>
    <div class="row">
        <div class="portfolio_container clearfix">
            <div class="element design span4">
                <div class="portfolio_img">
                    <img src="images/portfolio_img_1.jpg" class="pf_img" alt="img"/>
                    <article class="portfolio_link">
                        <a href="images/portfolio_img_1.jpg" class="example7 zoom"
                        title="<h3>BRAND</h3> Proin fringilla nunc tristique nulla a mattis. Proin fringilla turpis sit amet nunc 
                        blandit vehicula. Nulla Integer lacinia tristique nulla a mattis. Proin fringilla turpis sit amet nunc 
                        blandit vehicula. Nulla quis purus ut erat luctus congue. Vivamus consequat congue 
                        tristiqueDonec nec ligula turpis"> 
                        <img src="images/zoom_icon.png" alt="img"/>
                        </a>
                    </article>
                </div>
                <article class="content">
                    <!--<span class="strip"></span>!-->
                    <h4>BRAND?</h4>
                    <!--<p>Proin fringilla</p>!-->
                </article>
            </div>   
            <div class="element website span4">
                <div class="portfolio_img">
                    <img src="images/portfolio_img_2.jpg" class="pf_img" alt="img"/>
                    <article class="portfolio_link">
                        <a href="images/portfolio_img_2.jpg" class="example7 zoom"
                        title="<h3>BLACKBERRY</h3> Turpis sit amet nunc tristique nulla a mattis. Proin fringilla turpis sit amet nunc 
                        blandit vehicula. Nulla Integer lacinia tristique nulla a mattis. Proin fringilla turpis sit amet nunc 
                        blandit vehicula. Nulla quis purus ut erat luctus congue. Vivamus consequat congue 
                        tristiqueDonec nec ligula congue"> 
                        <img src="images/zoom_icon.png" alt="img"/>
                        </a>
                    </article>
                </div>
```

JQUERY

  // ====================== Filtering ======================

```
_filter : function( jQueryatoms ) {
  var filter = this.options.filter === '' ? '*' : this.options.filter;

  if ( !filter ) {
    return jQueryatoms;
  }

  var hiddenClass    = this.options.hiddenClass,
      hiddenSelector = '.' + hiddenClass,
      jQueryhiddenAtoms   = jQueryatoms.filter( hiddenSelector ),
      jQueryatomsToShow   = jQueryhiddenAtoms;

  if ( filter !== '*' ) {
    jQueryatomsToShow = jQueryhiddenAtoms.filter( filter );
    var jQueryatomsToHide = jQueryatoms.not( hiddenSelector ).not( filter ).addClass( hiddenClass );
    this.styleQueue.push({ jQueryel: jQueryatomsToHide, style: this.options.hiddenStyle });
  }

  this.styleQueue.push({ jQueryel: jQueryatomsToShow, style: this.options.visibleStyle });
  jQueryatomsToShow.removeClass( hiddenClass );

  return jQueryatoms.filter( filter );
},
```

**Comment by atuf78** [comment]
Hi, 

I am using a plugin call Wordpress Social Stream, these guys using Isotope, what i need to ask
1)Is there anyway i can see only 4 feeds instead of all feeds first time? here is the code hope you understand

sortstream($container, 'asc');

  jQuery('li .section-thumb img, li .section-text img', $container).css('opacity', 0).show().fadeTo(800, 1);
  $container.isotope('layout');

  function sortstream(obj, d) {
    var $l = jQuery('li', obj);
    $l.sort(function(a, b) {
      var keyA = parseInt(jQuery(a).attr('rel'), 10),
        keyB = parseInt(jQuery(b).attr('rel'), 10);
      if (d == 'asc') {
        return (keyA > keyB) ? 1 : -1;
      } else {
        return (keyA < keyB) ? 1 : -1;
      }
      return 0;
    });
    jQuery.each($l, function(index, row) {
      obj.append(row);
    });
    return;
  }
});

2) the second question is related to first question how can i show random feeds on first time but still only 4 feeds

Here is the plugin link http://codecanyon.net/item/wordpress-social-stream/2201708

Please help me out :(

Thanks in advance

**Comment by bsmith64** [comment]
I had the same problem with Isotope's masonry layout on load until I came across Desandro's solution. My masonry layout was positioning property once I applied a filter but not on document load. The solution was creating a separate class on the outer div on each item and using that class as the filter within a$(window).load(function());

`jQuery(window).load(function($){
  jQuery('.itemcontainer').isotope({
    itemSelector: '.item-box-outer',
    layoutMode: 'masonry',
     masonryHorizontal: {
      rowHeight: 300
    },
    filter: '.item-filter-all'
  });
});`

Thanks for the solution, cheers!
